OneCommunityGlobal/HighestGoodNetworkApp
# Jerry - See Only Weekly Summaries Reports Tab

# Description
<img width="732" alt="see only weekly summaries reports tab - description" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/55968519/dc049fe2-3f8a-44d2-9c72-f80f6ac1ced6">

The main motivation of this pull request is to allow an Owner account to assign the "_See Only Weekly Summaries Reports Tab_" to an account that does not have that permission by default (e.g., a Volunteer account).

## Related PRS (if any):
None

## Main changes explained:
- Conditionally render `Reports->Weekly Summaries Reports` option in the header for a Volunteer account that had the “_See Only Weekly Summaries Reports Tab_” permission added by an Owner account.
- Similarly, an Owner account can remove a Volunteer account’s “_See Only Weekly Summaries Reports Tab_” permission and the `Reports->Weekly Summaries Reports` option in the header will no longer be visible/accessible.
- Fixed permission title to “_See Only Weekly Summaries Reports Tab_” in `UserRoleTab.jsx`
- Fixed typo in `languages/en/ui.js` -- “Weekly Summaries Report” was missing a ’s’ in “Report”. It is now “Weekly Summaries Reports"
- Added `RoutePermissions.seeOnlyWeeklySummariesReports` to the `routePermissions` prop of the `ProtectedRoute` component (`src/routes.js`).
- Added “seeOnlyWeeklySummariesReports” key-value pair to `RoutePermissions` in `src/utils/routePermissions.js`
- Updated the title on the Permissions Management Page to "_See Only Weekly Summaries Reports Tab_" and updated the "i" description too to: _Makes the "Other Links" -> "Reports" button and ONLY the "Weekly Summaries Reports" option appear/accessible_.


## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Log in as Owner -> Other LInks -> Permissions Management -> Manage User Permissions
4. Enter the username of an account that does not have the “See Weekly Summary Reports Tab” permission by default (i.e., a Volunteer account). Click on Submit.
5. Login the Volunteer account (if already logged in, make sure to log out and log back in). The “Reports” button should appear with ONLY the “Weekly Summaries Reports” option visible/accessible.
6. The Owner account can also remove the “See Weekly Summary Reports Tab” permission. Click on Submit. Log into the Volunteer account and see that the "Reports" button is no longer visible/accessible.
7. **Note**: There is a typo in “Weekly Summaries Report” option where there is a missing ’s’ in “Report”. It should now be “Weekly Summaries Reports”. Also the title on the Permissions Management Page to is updated to "_See Only Weekly Summaries Reports Tab_". And the "i" description is also updated to: _Makes the "Other Links" -> "Reports" button and ONLY the "Weekly Summaries Reports" option appear/accessible_.


## Screenshots or videos of changes:

Development Branch:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/55968519/3c8c7a66-b723-4118-892c-0a747bde6e7d

Current Branch:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/55968519/eba47976-d7fd-4842-bfee-13cf0beec8e4

## Note:
Adding and removing the "_See Only Weekly Summaries Reports Tab_" to/from an account that has the permission to access the Weekly Summaries Reports page by default does not have any affect. This is because the code in this pull request conditionally renders the "_Weekly Summaries Reports_" tab based on whether or not the "seeOnlyWeeklySummariesReports" permission is present in `userProfiles.permissions.frontPermissions`. Since a Volunteer account does not have the permission to view the Weekly Summaries Reports page by default, this pull request will check if "seeOnlyWeeklySummariesReports" is inside `userProfiles.permissions.frontPermissions` and grant the volunteer account access to the Weekly Summaries Reports page.

**Edit (Aug 29, 2023)**: Due to a general problem, your Owner account may have the “Other Links” button in the header disappeared. You can try Empty Cache and Hard Reset (instructions in the screenshot below):

<img width="315" alt="Screenshot 2023-08-29 at 3 07 09 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/55968519/7d407340-06ee-4f26-868d-09e867577a5d">

Or create a new Owner account. I had issues creating new accounts in localhost, so I created a new Owner account in the HGN dev app. Hopefully your new Owner account will have the “Other Links” button in the header.
